diff --git a/src/cmake/externalpackages.cmake b/src/cmake/externalpackages.cmake
index 70ac351e..f5b76382 100644
--- a/src/cmake/externalpackages.cmake
+++ b/src/cmake/externalpackages.cmake
@@ -155,9 +155,6 @@ if (NOT BZIP2_FOUND)
     set (BZIP2_LIBRARIES "")  # TODO: why does it break without this?
 endif ()
 
-checked_find_package (Freetype
-                   DEFINITIONS  -DUSE_FREETYPE=1 )
-
 checked_find_package (OpenColorIO
                       DEFINITIONS  -DUSE_OCIO=1 -DUSE_OPENCOLORIO=1
                       # PREFER_CONFIG
